src/module/actor/sheet/base.ts

```diff
@@ -157,18 +157,32 @@ abstract class ActorSheetPF2e<TActor extends ActorPF2e> extends ActorSheet<TActo
 
     protected prepareInventory(): SheetInventory {
         const sections: SheetInventory["sections"] = {
-            weapon: { label: game.i18n.localize("PF2E.InventoryWeaponsHeader"), type: "weapon", items: [] },
-            armor: { label: game.i18n.localize("PF2E.InventoryArmorHeader"), type: "armor", items: [] },
-            equipment: { label: game.i18n.localize("PF2E.InventoryEquipmentHeader"), type: "equipment", items: [] },
-            consumable: { label: game.i18n.localize("PF2E.InventoryConsumablesHeader"), type: "consumable", items: [] },
-            treasure: { label: game.i18n.localize("PF2E.InventoryTreasureHeader"), type: "treasure", items: [] },
-            backpack: { label: game.i18n.localize("PF2E.InventoryBackpackHeader"), type: "backpack", items: [] },
+            weapon: {
+                label: game.i18n.localize("PF2E.Actor.Inventory.Section.WeaponsAndShields"),
+                types: ["shield", "weapon"],
+                items: [],
+            },
+            armor: { label: game.i18n.localize("PF2E.InventoryArmorHeader"), types: ["armor"], items: [] },
+            equipment: { label: game.i18n.localize("PF2E.InventoryEquipmentHeader"), types: ["equipment"], items: [] },
+            consumable: {
+                label: game.i18n.localize("PF2E.InventoryConsumablesHeader"),
+                types: ["consumable"],
+                items: [],
+            },
+            treasure: { label: game.i18n.localize("PF2E.InventoryTreasureHeader"), types: ["treasure"], items: [] },
+            backpack: { label: game.i18n.localize("PF2E.InventoryBackpackHeader"), types: ["backpack"], items: [] },
         };
 
         for (const item of this.actor.inventory.contents.sort((a, b) => (a.sort || 0) - (b.sort || 0))) {
-            if (!objectHasKey(sections, item.type) || item.isInContainer) continue;
-            const category = item.isOfType("book") ? sections.equipment : sections[item.type];
-            category.items.push(this.prepareInventoryItem(item));
+            if (item.isInContainer) continue;
+            const section = objectHasKey(sections, item.type)
+                ? sections[item.type]
+                : item.isOfType("book")
+                  ? sections.equipment
+                  : item.isOfType("shield")
+                    ? sections.weapon
+                    : null;
```

Comment:
Should this be a search for the section's types instead? Of course after adding book to the equipment types.

Reply:
That would be better, yes